src/lib/protocols/credentials/messages/CredentialOfferMessage.ts

```diff
@@ -0,0 +1,95 @@
+import { Equals, IsString } from 'class-validator';
+import { Expose } from 'class-transformer';
+
+import { AgentMessage } from '../../../agent/AgentMessage';
+import { MessageType } from './MessageType';
+
+interface CredentialPreviewOptions {
+  attributes: CredentialPreviewAttribute[];
+}
+
+export class CredentialPreview {
+  public constructor(options: CredentialPreviewOptions) {
+    this.attributes = options.attributes;
+  }
```

Comment:
You should check if options is defined or provide a default value (` = {}`) because class transformer calls the constructor.